### PR TITLE
refactor: split `lint` script into `lint`, `compile` and `test`

### PR DESCRIPTION
**Pull request by edouard-lopez**
update CI steps

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,9 @@
     "terminal-screenshot": "out/src/cli.js"
   },
   "scripts": {
-    "lint": "jest && prettier . --list-different && tsc"
+    "lint": "prettier . --list-different && tsc",
```
Comment (OmarTawfik) [inline package.json]: Left over?

```suggestion
    "lint": "prettier . --list-different",
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -12,6 +12,8 @@ jobs:
           node-version: 16
       - run: yarn --frozen-lockfile
       - run: yarn lint
```
Comment (OmarTawfik) [inline .github/workflows/ci.yaml:15]: super nit: I suggest running `tsc`, `jest`, then `prettier` in that order, to make sure the most critical issues are fixed first.

**Review by OmarTawfik — CHANGES_REQUESTED** [review]
Looks great! Thank you.
I left a few suggestions (spelling, duplicate commands). Please let me know what you think.

File: package.json [hunk 3]
```diff
@@ -11,7 +11,9 @@
     "terminal-screenshot": "out/src/cli.js"
   },
   "scripts": {
-    "lint": "jest && prettier . --list-different && tsc"
+    "lint": "prettier . --list-different",
+    "build": "tsc",
```
Comment (edouard-lopez) [inline package.json:15]: @OmarTawfik Note, that I renamed `compile` to `build` which is more common

Comment (OmarTawfik) [reply package.json:15]: Sounds good.

**Review by OmarTawfik — APPROVED** [review]
Thank you for the refactoring!

**Pull request merged**